src/local_sensitivity/concrete_solve.jl

```diff
@@ -47,7 +50,12 @@ function _concrete_solve_adjoint(prob,alg,sensealg::AbstractAdjointSensitivityAl
   if haskey(kwargs, :callback_adj)
     kwargs_adj = merge(kwargs_adj, NamedTuple{(:callback,)}( [get(kwargs, :callback_adj, nothing)] ))
   end
-  sol = solve(_prob,alg,args...;save_start=true,save_end=true,kwargs...)
+
+  if ischeckpointing(sensealg)
+    sol = solve(_prob,alg,args...;save_start=true,save_end=true,saveat=saveat,kwargs...)
+  else

```

Comment:
```suggestion
  else
    isempty(saveat) || @warn "The algorithm has checkpointing disabled, but `saveat` is non-empty. Please either enable checkpointing to save memory usage or remove `saveat`"
```

Reply:
I don't think we want to warn, since small problems with saveat is probably the most common use for this function, and so this would be warning on the 99% use case. But we really really need to document this well.